Update Azure quotas for additional node families

micheleRP opened the pull request:
## Description
This adds to Azure quota prereqs + updates the style of the prereqs to make it a bit easier to digest. 

Resolves https://redpandadata.atlassian.net/browse/DOC-538
Review deadline: Friday Oct 18

## Page previews
[Create a BYOC Cluster on Azure - Prereqs](https://deploy-preview-92--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-azure/)

<!--- add your page preview here. 
A simple way to do it is to open the link generated by Netlify bot + file path. Remember to remove page, module, and the .adoc extension.
A preview looks like this
https://deploy-preview-487--redpanda-docs-preview.netlify.app/current/manage/node-management/
https://deploy-preview-<PR-NUMBER>--redpanda-docs-preview.netlify.app/<VERSION>/<PATH-TO-FILE-WITHOUT-ADOC>
-->

## Checks

- [ ] New feature
- [X] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

ligfx commented:
Adding `Standard Dv3-series vCPUs` to the list and noting what each instance type is used for makes sense to me.

I'm not sure I understand the point of making the top-level items checkboxes (and it feels a bit busy).

I wonder if it might make sense to pull the quotas out into their own heading, similar to the SKU restrictions (or a combined section with them?). We'll eventually add more detail there when we release higher tiers anyways.

c4milo commented on modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:
```diff
@@ -16,17 +16,20 @@ Confirm you have a minimum version of Redpanda `rpk` v24.1. See xref:reference:r
+- [ ] **Role**: The Azure user has the _Owner_ role in the subscription.
+- [ ] **Quota**: The subscription has enough total regional vCPUs and vCPUs per instance family in the region where you will use Redpanda. See the https://learn.microsoft.com/en-us/azure/quotas/view-quotas[Microsoft documentation^].
++
+* Standard LASv3-series vCPUs:  24 (3 Redpanda broker nodes)
```
We now deploy Kafka Connect too. @chapmanc will know more.

chapmanc replied:
If the value is disabled we do also deploy a kafka-connect node pool. The defaults are [here](https://github.com/redpanda-data/cloudv2/blob/9db1238416d98f9d852a2df845f46589795f0d16/apps/controlplane-api/defaults/defaulter.go#L618-L632) or specified by the `tier` and `connectors_machine_type` in the install pack: https://github.com/redpanda-data/cloudv2/blob/9db1238416d98f9d852a2df845f46589795f0d16/install-pack/24.2.yml#L4892

micheleRP replied:
Thanks @chapmanc: Does [this section](https://deploy-preview-92--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-azure/#check-azure-quota) look correct?

micheleRP commented:
> Adding `Standard Dv3-series vCPUs` to the list and noting what each instance type is used for makes sense to me.
> 
> I'm not sure I understand the point of making the top-level items checkboxes (and it feels a bit busy).
> 
> I wonder if it might make sense to pull the quotas out into their own heading, similar to the SKU restrictions (or a combined section with them?). We'll eventually add more detail there when we release higher tiers anyways.

Thanks @ligfx: I moved quota into its own section and removed the checkboxes. Please eyeball the [prereqs ](https://deploy-preview-92--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-azure/#prerequisites)again and let me know if it's clear!

ligfx commented:
> Thanks @ligfx: I moved quota into its own section and removed the checkboxes. Please eyeball the [prereqs ](https://deploy-preview-92--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-azure/#prerequisites)again and let me know if it's clear!

LGTM!

asimms41 commented on modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:
````diff
@@ -40,20 +40,17 @@ If it is not registered, run:
 ```
 
-* Feature: The subscription must be registered for Microsoft.Compute/EncryptionAtHost. See the https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites[Microsoft documentation^].
+* **Feature**: The subscription must be registered for Microsoft.Compute/EncryptionAtHost. See the https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites[Microsoft documentation^].
````
```suggestion
* **Feature**: The subscription must be registered for EncryptionAtHost/Microsoft.Compute. See the https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites[Microsoft documentation^].
Just suggested a change in order to match the order in the command below.
```

ligfx replied:
The correct name is `Microsoft.Compute/EncryptionAtHost`, which is what Azure calls it in error messages. The command below could be reordered if needed.

micheleRP replied:
changed command to: 
`az feature register --namespace Microsoft.Compute --name EncryptionAtHost`

asimms41 reviewed (approved):
LGTM. Just one small suggestion.
